ld/emultempl/riscvelf.em

```diff
@@ -39,7 +43,16 @@ riscv_elf_before_allocation (void)
   else
     ENABLE_RELAXATION;
 
-  link_info.relax_pass = 2;
+  /* check if all of input files are binary blob.
+     If true, I skip relaxation pass. */
+  for (abfd = link_info.input_bfds; abfd != NULL; abfd = abfd->link.next) {
+    if (abfd->xvec != &binary_vec) {
+      all_inputs_binary = FALSE;
+      break;
+    }
+  }
+
+  link_info.relax_pass = all_inputs_binary == TRUE? 0: 2;
```

Comment:
Hi, Andrew,
I wrote a original fix for freebsd binutils branch.
I also think this is a kind of dirty hack.  
The riscv-gnu-toolchain-freebsd branch does not contain all of binutils sources. So I chose to fix anything as much as possible in sources left on the branch. If the bsd suppose is fully merged to the mainline, I think we can implement it in more cleaner way.

> In relax_section, there's a check that bails out if reloc_count == 0. Why isn't that triggered for binary sections?
Although I don't remember details exactrly, I suppose the value of link_info.relocatable is checked in bfd_generic_relax_sections() in bfd/reloc.c. and its called before the vail out by reloc_count ==0.
# Now I come back to my family home without bringing my development environment and memo.

Anyway, I'll try to write new patch that is for the riscv binutils mainline after I go back to my house next week.